python/deltalake/writer.py

```diff
@@ -162,15 +228,27 @@ def write_deltalake(
         configuration: A map containing configuration options for the metadata action.
         overwrite_schema: If True, allows updating the schema of the table.
         storage_options: options passed to the native delta filesystem. Unused if 'filesystem' is defined.
-        partition_filters: the partition filters that will be used for partition overwrite.
+        predicate: When using `Overwrite` mode, replace data that matches a predicate. Only used in rust engine.
+        partition_filters: the partition filters that will be used for partition overwrite. Only used in pyarrow engine.
         large_dtypes: If True, the data schema is kept in large_dtypes, has no effect on pandas dataframe input
     """
-
     table, table_uri = try_get_table_and_table_uri(table_or_uri, storage_options)
+    if table is not None:
+        storage_options = table._storage_options or {}
+        storage_options.update(storage_options or {})
 
-    # We need to write against the latest table version
-    if table:
         table.update_incremental()
+        if table.protocol().min_writer_version > MAX_SUPPORTED_WRITER_VERSION:
```

Comment:
we should use the global `PROTOCOL` instance (specifically `can_write_to`) from the transaction module to check if we can write to a table.

Reply:
@roeap I'll take a look!  I'll probably push this check down to the pyarrow engine only since we still need it but for rust engine we can check with the protocol instance


@wjones127 With respect to all these changes around the protocol support and soon having check constraints, should we support this in pyarrow engine, or we can communicate to users, moving forward rust engine will get all the new features?

Reply:
@roeap ah it seems, it's already there in the `fn check_preconditions`, which is executed when the builder is converted IntoFuture

Reply:
true :), then we should just delete that check here.

Reply:
Yup :) Removed it, check its only done for pyarrow writer engine in python.